pandas/core/resample.py

```diff
@@ -859,6 +861,11 @@ def interpolate(
         Interpolate values according to different methods.
         """
         result = self._upsample("asfreq")
+        if isinstance(result.index, DatetimeIndex):
+            obj = self._selected_obj
+            tmp = concat([obj, result]).sort_index().interpolate(method='time')
+            tmp = tmp[result.index]
+            result[...] = tmp[~tmp.index.duplicated(keep='first')]
```

Comment:
you are not following standard code guideines, e.g. use " instead of '.

run `> black pandas` this will automatically fix these issues for you.